dashboard/app/models/pd/teacher_application.rb

```diff
@@ -86,6 +86,22 @@ def validate_required_application_fields
     end
   end
 
+  validate :primary_email_must_match_user_email
+  def primary_email_must_match_user_email
+    if user && user.hashed_email != Digest::MD5.hexdigest(primary_email)
+      account_settings_link = ActionController::Base.helpers.link_to('account settings', '/users/edit')
+      errors.add(
+        :primary_email,
+        "must match your login. If you want to use this email instead, first update it in #{account_settings_link}."
+      )
+    end
+  end
+
+  after_create :ensure_user_is_a_teacher
+  def ensure_user_is_a_teacher
+    user.update!(user_type: User::TYPE_TEACHER, email: primary_email) if user.email.blank?
```

Comment:
Do you want to throw here if the `User` is invalid after the update? My hope is that we do (I'd like to see the issue surfaced in HB). Realize that (IIRC) we have a bunch of invalid users (beyond missing email).